stac/src/main/scala/StacLicense.scala

```diff
@@ -0,0 +1,41 @@
+package geotrellis.server.stac
+
+import cats.implicits._
+import io.circe._
+
+sealed trait StacLicense {
+  val name: String
+}
+
+final case class Proprietary() extends StacLicense {
```

Comment:
Perhaps just a `case object` if non-parametric creation

Reply:
I'm slightly surprised this works

```scala
@ trait Foo {
    val x: Int
  } 
defined trait Foo

@ case object Bar extends Foo {
    val x: Int = 3
  } 
defined object Bar
```